undocked/tests/sample/ext/drv/sample_ext.c

```diff
@@ -259,23 +267,24 @@ _sample_ebpf_extension_program_info_provider_attach_client(
     *provider_binding_context = NULL;
     *provider_dispatch = NULL;
 
-    program_info_client = (sample_ebpf_extension_program_info_client_t*)ebpf_allocate(
-        sizeof(sample_ebpf_extension_program_info_client_t));
-
+    program_info_client = ExAllocatePoolUninitialized(
```

Comment:
Why the change to use ExAllocate* instead of ebpf_allocate*?   Functionally this is ok, but it means the sample will be less useful for a user-mode case in the future.

Reply:
We want undocked code to be able to use cxplat APIs.  The point is that all extensions, not just docked ones,  should be able to get the benefit of fuzz testing, etc.

Reply:
Hi Dave, agree that undocked code can use cxplat APIs. If someone is developing an extension in a separate repo, they can use cxplat for development. In this case though, the main intention is to ensure the undocked projects only need the NuGet package (and not use any "internal" header file from the ebpf repo - this PR is fixing multiple eBPF headers which are published in the NuGet package).

My main concern is that including an "internal" file or lib may inadvertently pull in some other dependencies, and it makes it harder to validate that the undocked component is still truly undocked.

Reply:
I can make the changes to use `cxplat_allocate` and that will require linking to `cxplat_winkernel.lib`. This lib is being built as an internal lib in ebpf repo (not added in the eBPF NuGet package), even though it uses source files from usersim submodule to build the lib. 

Another thing I just noticed is that all of netebpfext code today calls `ExAllocatePoolUninitialized` (and nowhere calls `ebpf_allocate`).

Reply:
Changed to using cxplat APIs.